src/Provider.js

```diff
@@ -114,6 +120,9 @@ class Provider extends Component {
   }
 
   componentWillUnmount() {
+    if (this.timerId.length) {
```

Comment:
There's no need for this check here, the `forEach` will not be executed if the array is empty.

Also, if you just want to execute a function for each element you can simply pass that function as the parameter.

So we can simplify this 3 lines to just `this.timerId.forEach(clearTimeout)`.

Reply:
Done with the recommendation